### PR TITLE
PR: Ability to delete posts based on number of comments #52

### LINKED ISSUE #52: Ability to delete posts based on number of comments
@esudalaimadan 

Use the following content for the module.

- Title => Delete Posts based on the number of comments
- Select the comment settings based on which posts should be deleted
- Delete posts that have comments {dropdown} {textbox}

### PR DESCRIPTION
**Pull request by esudalaimadan**
Adds Ability to delete posts based on number of comments

Fix #52

### CODE REVIEW

**Comment by esudalaimadan** [comment]
@sudar 
1. Posts with private and draft status should be deleted or not?
2. Need your help in displaying the module in the right place. Currently, it is getting displayed at the bottom of the page.

File: assets/js/src/delete-posts-by-comments.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/**
+ * JavaScript for delete posts by comments module of Bulk Delete Plugin.
+ *
+ * @author Sudar <https://bulkwp.com>
+ * @since 6.1.0
```
Comment (sudar) [inline assets/js/src/delete-posts-by-comments.js]: Let's change the version to `6.0.0`. I am planning to release it as part of `6.0.0` itself.

File: include/Core/BulkDelete.php [hunk 2]
```diff
@@ -348,6 +349,7 @@ private function get_delete_posts_admin_page() {
 		$posts_page->add_module( new DeletePostsByURLModule() );
 		$posts_page->add_module( new DeletePostsByRevisionModule() );
 		$posts_page->add_module( new DeletePostsByStickyPostModule() );
+		$posts_page->add_module( new DeletePostsByCommentsModule() );
```
Comment (sudar) [inline include/Core/BulkDelete.php]: Add this module before Delete by URL module.

File: include/Core/Posts/Modules/DeletePostsByCommentsModule.php [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+	 */
+	protected function initialize() {
+		$this->item_type     = 'posts';
+		$this->field_slug    = 'comments';
+		$this->meta_box_slug = 'bd_by_comments';
+		$this->action        = 'delete_posts_by_comments';
+		$this->cron_hook     = 'do-bulk-delete-comments';
+		$this->scheduler_url = 'http://bulkwp.com/addons/scheduler-for-deleting-posts-by-comments/?utm_source=wpadmin&utm_campaign=BulkDelete&utm_medium=buynow&utm_content=bd-st';
+		$this->messages      = array(
+			'box_label'  => __( 'By Post Comments', 'bulk-delete' ),
```
Comment (sudar) [inline include/Core/Posts/Modules/DeletePostsByCommentsModule.php]: Let's change this label to `By Comment count`.

File: include/Core/Posts/Modules/DeletePostsByCommentsModule.php [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+		<!-- Comments start-->
+		<fieldset class="options">
+			<table class="optiontable">
+				<tr>
+					<td scope="row" colspan="2">
+						<?php _e( 'Delete posts that have comments', 'bulk-delete' ); ?>
+					</td>
+					<td>
+						<select name="smbd_<?php echo esc_attr( $this->field_slug ); ?>_count_op" id="smbd_<?php echo esc_attr( $this->field_slug ); ?>_count_op">
```
Comment (sudar) [inline include/Core/Posts/Modules/DeletePostsByCommentsModule.php]: We have a helper function called `render_number_comparison_operators` in `Renderer` class. Let's use that here.

File: assets/js/src/delete-posts-by-comments.js (hunk 1 above)
Comment (esudalaimadan) [reply assets/js/src/delete-posts-by-comments.js]: Fixed.

File: include/Core/BulkDelete.php (hunk 2 above)
Comment (esudalaimadan) [reply include/Core/BulkDelete.php]: Fixed.  But still the module appears at the bottom of the page.  Kindly look into it.

File: include/Core/Posts/Modules/DeletePostsByCommentsModule.php (hunk 3 above)
Comment (esudalaimadan) [reply include/Core/Posts/Modules/DeletePostsByCommentsModule.php]: Fixed.

File: include/Core/Posts/Modules/DeletePostsByCommentsModule.php (hunk 4 above)
Comment (esudalaimadan) [reply include/Core/Posts/Modules/DeletePostsByCommentsModule.php]: Option value differs(Please refer slack for more detailed info).  Kindly advice.

File: include/Core/Posts/Modules/DeletePostsByCommentsModule.php [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+namespace BulkWP\BulkDelete\Core\Posts\Modules;
+
+use BulkWP\BulkDelete\Core\Posts\PostsModule;
+
+defined( 'ABSPATH' ) || exit; // Exit if accessed directly.
+
+/**
+ * Delete Posts by Comments Module.
+ *
+ * @since 6.1.0
```
Comment (sudar) [inline include/Core/Posts/Modules/DeletePostsByCommentsModule.php]: Change this (and the version in all other places) to `6.0.0`

File: include/Core/Posts/Modules/DeletePostsByCommentsModule.php (hunk 4 above)
Comment (sudar) [reply include/Core/Posts/Modules/DeletePostsByCommentsModule.php]: @esudalaimadan 

Let's change the helper function to use the values that you are using.

After making that change, find the place where the helper function is currently being used and then change the values in that module as well.

I did a quick search and we are using the helper function in only one other place.

File: include/Core/Posts/Modules/DeletePostsByCommentsModule.php (hunk 5 above)
Comment (esudalaimadan) [reply include/Core/Posts/Modules/DeletePostsByCommentsModule.php]: Fixed.

**Pull request merged**